### PR TITLE
Pia 2146 network service

### PR DESCRIPTION
**Pull request by zladzeyka**
Capture SDK:
   - Add Screen API support with custom networking:
    1.   Add default implementation`GiniCaptureNetworkService`;
    2.   Add initializer in GiniCapture networking with non-optional `GiniCaptureNetworkService` parameter.
    
 Capture SDK Example:
  - Fix code style warnings and refactor protocol implementation into the extensions.

### CODE REVIEW

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/DocumentService.swift [hunk 1]
```diff
@@ -27,34 +52,57 @@ public final class DocumentService: DocumentServiceProtocol {
...
+        }
     }
     
     public func cancelAnalysis() {
         if let compositeDocument = document {
-            delete(compositeDocument)
+            captureNetworkService.delete(document: compositeDocument) {[weak self] result in
+                switch result {
+                case .success:
+                    self?.analysisCancellationToken?.cancel()
```
Comment (a-szotyori) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/DocumentService.swift:75]: Seeing the old code and the new one side-by-side I noticed, that the old code always invoked the following three lines:
```
self?.analysisCancellationToken?.cancel()
self?.analysisCancellationToken = nil
self?.document = nil
```

It did not check whether there was a document or wether deletion succeeded or not. I recommend moving these lines back outside of the callback and after the if block.

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/DocumentService.swift [hunk 2]
```diff
@@ -27,34 +52,57 @@ public final class DocumentService: DocumentServiceProtocol {
...
     
     public func remove(document: GiniCaptureDocument) {
         if let index = partialDocuments.index(forKey: document.id) {
             if let document = partialDocuments[document.id]?
                 .document {
-                delete(document)
+                captureNetworkService.delete(document: document) {[weak self] result in
+                    switch result {
+                    case .success:
+                        self?.partialDocuments.remove(at: index)
```
Comment (a-szotyori) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/DocumentService.swift:92]: Here, too, calling the `partialDocuments.remove(at: index)` did not depend on the deletion result or whether there was a partial document for the `document.id`.

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/DocumentServiceProtocol.swift [hunk 3]
```diff
@@ -18,35 +18,13 @@ public protocol DocumentServiceProtocol: AnyObject {
     var analysisCancellationToken: CancellationToken? { get set }
     
     func cancelAnalysis()
-    func remove(document: GiniCaptureDocument)
-    func resetToInitialState()
+    func remove(document: GiniCaptureDocument) // delete
```
Comment (a-szotyori) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/DocumentServiceProtocol.swift:21]: Whoops. I committed these trailing comments by accident.

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/DocumentService.swift [hunk 4]
```diff
@@ -14,11 +14,36 @@ public final class DocumentService: DocumentServiceProtocol {
...
     public var analysisCancellationToken: CancellationToken?
     public var metadata: Document.Metadata?
-    var documentService: DefaultDocumentService
+    
+    var captureNetworkService: GiniCaptureNetworkService
     
     public init(lib: GiniBankAPI, metadata: Document.Metadata?) {
         self.metadata = metadata
-        self.documentService = lib.documentService()
+        self.captureNetworkService = DefaultCaptureNetworkService(lib: lib)
```
Comment (a-szotyori) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/DocumentService.swift:22]: Nice! Very elegantly solved to keep the internal changes hidden from the outside and avoid changing existing code. 👏

**Review by a-szotyori — CHANGES_REQUESTED** [review]
Nice! I just noticed behavior changes when seeing the old and new code side-by-side.

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/DocumentServiceProtocol.swift (hunk 3 above)
Comment (zladzeyka) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/DocumentServiceProtocol.swift]: I'll remove the comments

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/DocumentService.swift (hunk 2 above)
Comment (zladzeyka) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Networking/DocumentService.swift:92]: Ok, it makes sense.

**Review by a-szotyori — APPROVED** [review]

**Pull request merged**